python/cudf/cudf/io/csv.py

```diff
@@ -61,9 +61,17 @@ def read_csv(
         path_or_data=filepath_or_buffer,
         compression=compression,
         iotypes=(BytesIO, StringIO),
+        byte_ranges=[byte_range] if byte_range else None,
+        clip_dummy_buffer=True if byte_range else False,
         **kwargs,
     )
```

Comment:
**Notes**:

- When `byte_range` is specified in a `read_csv` call, we only need to transfer **that** byte range from remote storage.
- By default, the byte-ranges that are transferred from remote storage will be coppied into a local dummy buffer.  We call this a dummy buffer, because it is likely to contain many "empty" bytes that libcudf will ultimately ignore
- We use `clip_dummy_buffer=True` to avoid the actual allocation of "empty" bytes when `byte_range=` is specified. This option informs `get_filepath_or_buffer` that the local dummy buffer does not need to be the size of the entire remote file, and can be clipped down to the exact `byte_range` size.
- When `clip_dummy_buffer=True`, the `byte_range` argument passed down to libcudf must be adjusted to a zero offset (see code block below).
- The `clip_dummy_buffer=True` optimization cannot be used for parquet (for now), because the footer metadata includes many specific column-chunk offesets.